### PR TITLE
Update publish { ... } to use more idiomatic DSL

### PR DESCRIPTION
**Pull request by jakeouellette**

### CODE REVIEW

File: src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy [hunk 1]
```diff
@@ -38,7 +38,7 @@ class IntelliJPlugin implements Plugin<Project> {
     @Override
     def void apply(Project project) {
         project.getPlugins().apply(JavaPlugin)
-        def intellijExtension = project.extensions.create(EXTENSION_NAME, IntelliJPluginExtension)
```
Comment (mfurtak) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy:41]: Remove `EXTENSION_NAME` constant if unused?

File: src/main/groovy/org/jetbrains/intellij/IntelliJPluginExtension.groovy [hunk 2]
```diff
@@ -24,36 +26,13 @@ class IntelliJPluginExtension {
         return runClasspath
     }
 
-    @SuppressWarnings("GroovyUnusedDeclaration")
-    def publish(Closure c) {
-        publish.with(c)
-    }
-
     public static class Publish {
```
Comment (mfurtak) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPluginExtension.groovy]: I don't know how relevant it is, but my preference would be to have this be its own top-level class unless it somehow benefits from the nesting. If you're trying to minimize changes to their stuff, no problem

**Comment by mfurtak** [comment]
LGTM :sheep: at will

File: src/main/groovy/org/jetbrains/intellij/IntelliJPluginExtension.groovy (hunk 2 above)
Comment (jakeouellette) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPluginExtension.groovy]: Agreed.

File: src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy (hunk 1 above)
Comment (jakeouellette) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy:41]: Agreed.

**Pull request merged**